.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,65 @@
+name: Setup Python and install requirements
+inputs:
+  python-version:
+    required: true
+  architecture:
+    default: x64
+runs:
+  using: composite
+  steps:
+    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
+    - name: Create temporary requirements.txt
```

Comment:
Any reason for this? I think we can pin setuptools in the main file.

Reply:
This is required to use `setup-python`'s caching.